internal/controllers/reconciler.go

```diff
@@ -442,7 +442,7 @@ func (r *Reconciler) createOrUpdateDeployment(ctx context.Context, deploy *appsv
 	op, err := controllerutil.CreateOrUpdate(ctx, r.Client, deploy, func() error {
 		// TODO consider managing labels and annotations using CRD
```

Comment:
@ebaron would this TODO be addressed by these changes?

Reply:
Yes, I think that TODO is now covered by this PR and can be deleted if you don't mind @OpenGuidou

Reply:
Done